File: remote/src/main/java/com/kelsos/mbrc/mappers/MapperUtils.java
package com.kelsos.mbrc.mappers;

import com.kelsos.mbrc.dao.AlbumDao;
import com.kelsos.mbrc.dao.ArtistDao;
import com.kelsos.mbrc.dao.CoverDao;
import com.kelsos.mbrc.dao.GenreDao;
import java.util.List;
import rx.Observable;

public class MapperUtils {
  private MapperUtils() {
    //no instance
  }

  static CoverDao getCoverById(int id, List<CoverDao> data) {
    return Observable.from(data).filter(o -> o.getId() == id).toBlocking().firstOrDefault(null);
  }

  static ArtistDao getArtistById(int id, List<ArtistDao> data) {
    return Observable.from(data).filter(o -> o.getId() == id).toBlocking().firstOrDefault(null);
  }

  static AlbumDao getAlbumById(int id, List<AlbumDao> data) {
    return Observable.from(data).filter(o -> o.getId() == id).toBlocking().firstOrDefault(null);
  }

  static GenreDao getGenreById(int id, List<GenreDao> data) {
    return Observable.from(data).filter(o -> o.getId() == id).toBlocking().firstOrDefault(null);
  }
}

File: mbrc-lib/src/main/kotlin/com/kelsos/mbrc/mappers/MapperUtils.kt
package com.kelsos.mbrc.mappers

import com.kelsos.mbrc.dao.AlbumDao
import com.kelsos.mbrc.dao.ArtistDao
import com.kelsos.mbrc.dao.CoverDao
import com.kelsos.mbrc.dao.GenreDao
import rx.Observable

object MapperUtils {

    internal fun getCoverById(id: Long, data: List<CoverDao>): CoverDao {
        return Observable.from(data).filter {
            it.id == id
        }.toBlocking().firstOrDefault(null)
    }

    internal fun getArtistById(id: Long, data: List<ArtistDao>): ArtistDao {
        return Observable.from(data).filter { it.id == id }.toBlocking().firstOrDefault(null)
    }

    internal fun getAlbumById(id: Long, data: List<AlbumDao>): AlbumDao {
        return Observable.from(data).filter { it.id == id }.toBlocking().firstOrDefault(null)
    }

    internal fun getGenreById(id: Long, data: List<GenreDao>): GenreDao {
        return Observable.from(data).filter { it.id == id }.toBlocking().firstOrDefault(null)
    }
}//no instance
